plugins/network-elements/ovs/src/com/cloud/network/element/OvsElement.java

```diff
@@ -488,50 +494,54 @@ public boolean applyPFRules(final Network network, final List<PortForwardingRule
             return true;
         }
 
-        DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
-        NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
-
-        return networkTopology.applyFirewallRules(network, rules, routers);
+        final DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
+        final NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
+        for (final DomainRouterVO domainRouterVO : routers) {
+            result = networkTopology.applyFirewallRules(network, rules, domainRouterVO);
```

Comment:
I already said why I didn't do it and  want to change the code. So, what's the point in going on and comment on every thing? We could just accept this and improve during the 4.7.0 release cycle.